jax/numpy/lax_numpy.py

```diff
@@ -2369,6 +2369,7 @@ def inner(a, b, precision=None):
 def outer(a, b, out=None):
   if out:
     raise NotImplementedError("The 'out' argument to outer is not supported.")
+  a, b = _promote_dtypes(a, b)
```

Comment:
Is this really needed? I would guess that `*` below would handle this.

Reply:
Yes, if we wish to have the same promotion behavior for `onp` and `np` arrays. My belief is that `jax.numpy` APIs should be consistent in their behavior regardless of which kind of array was passed to them.

Another option here would have been to add a `jit` decorator.